### PR TITLE
Vtcca 3209

### PR DESCRIPTION
**Pull request by habeeb01**

### CODE REVIEW

File: app/uk/gov/hmrc/voapropertylinking/config/GuiceModule.scala [hunk 1]
```diff
@@ -73,16 +73,17 @@ class GuiceModule(
...
+        "voa.myOrganisationsPropertyLinks"  -> "voa.resources.externalPropertyLink.myOrganisationsPropertyLinks.path",
+        "voa.myOrganisationsPropertyLink"   -> "voa.resources.externalPropertyLink.myOrganisationsPropertyLink.path",
+        "voa.myOrganisationsAgents"         -> "voa.resources.externalPropertyLink.myOrganisationsAgents.path",
+        "voa.myClientsPropertyLink"         -> "voa.resources.externalPropertyLink.myClientsPropertyLink.path",
+        "voa.myClientsPropertyLinks"        -> "voa.resources.externalPropertyLink.myClientsPropertyLinks.path",
+        "voa.createPropertyLink"            -> "voa.resources.externalPropertyLink.createPropertyLink.path",
+        "voa.createRepresentationRequest"   -> "voa.resources.authorisationManagementApi.createRepresentationRequest.path",
+        "voa.representationRequestResponse" -> "voa.resources.authorisationManagementApi.representationRequestResponse.path",
+        "voa.agentAppointmentChanges"       -> "voa.resources.organisationManagementApi.agentAppointmentChanges.path",
+        "voa.getAgentDetails"               -> "voa.resources.organisationManagementApi.getAgentDetails.path"
```
Comment (DanielConnelly) [inline app/uk/gov/hmrc/voapropertylinking/config/GuiceModule.scala]: myAgentsDetailsUrl

File: app/uk/gov/hmrc/voapropertylinking/controllers/PropertyRepresentationController.scala [hunk 2]
```diff
@@ -123,10 +125,20 @@ class PropertyRepresentationController @Inject()(
...
       case Some(agent) => Ok(Json.toJson(agent))
     }
   }
+
+  def agentAppointmentChanges(): Action[JsValue] = authenticated.async(parse.json) { implicit request =>
+    withJsonBody[AppointmentChangesRequest] { appointmentChangesRequest =>
+      organisationManagementApi
+        .agentAppointmentChanges(appointmentChangesRequest)
+        .map { response =>
+          Accepted(s"change id: ${response.appointmentChangeId}")
```
Comment (DanielConnelly) [inline app/uk/gov/hmrc/voapropertylinking/controllers/PropertyRepresentationController.scala]: Json in the response body please.

File: app/uk/gov/hmrc/voapropertylinking/connectors/modernised/OrganisationManagementApi.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import javax.inject.{Inject, Named}
+import uk.gov.hmrc.http._
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import uk.gov.hmrc.voapropertylinking.auth.RequestWithPrincipal
+import uk.gov.hmrc.voapropertylinking.http.VoaHttpClient
+import uk.gov.hmrc.voapropertylinking.models.modernised.agentrepresentation.{AgentDetails, AppointmentChangeResponse, AppointmentChangesRequest}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class OrganisationManagementApi @Inject()(
```
Comment (DanielConnelly) [inline app/uk/gov/hmrc/voapropertylinking/connectors/modernised/OrganisationManagementApi.scala]: ExternalOrganisationManagementApi so as to not confuse with the other API.

Comment (habeeb01) [reply app/uk/gov/hmrc/voapropertylinking/connectors/modernised/OrganisationManagementApi.scala]: i didn't know we had an "organisation-management-api" but I have now changed the name to ExternalOrganisationManagementApi

File: app/uk/gov/hmrc/voapropertylinking/config/GuiceModule.scala (hunk 1 above)
Comment (habeeb01) [reply app/uk/gov/hmrc/voapropertylinking/config/GuiceModule.scala]: done

File: app/uk/gov/hmrc/voapropertylinking/models/modernised/agentrepresentation/AppointAgent.scala [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.voapropertylinking.models.modernised.agentrepresentation
+
+import play.api.libs.json.{Json, OFormat}
+
+case class AppointAgent(agentRepresentativeCode: Long, scope: String) {}
```
Comment (DanielConnelly) [inline app/uk/gov/hmrc/voapropertylinking/models/modernised/agentrepresentation/AppointAgent.scala]: this object is missing the propertyLinkSubmissionId's.

File: test/uk/gov/hmrc/voapropertylinking/connectors/modernised/ExternalOrganisationManagementApiSpec.scala [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import basespecs.BaseUnitSpec
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import uk.gov.hmrc.voapropertylinking.models.modernised.agentrepresentation.{AgentDetails, AppointmentAction, AppointmentChangeResponse, AppointmentChangesRequest, AppointmentScope}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class ExternalOrganisationManagementApiSpec extends BaseUnitSpec {
+
```
Comment (darrell-plant) [inline test/uk/gov/hmrc/voapropertylinking/connectors/modernised/ExternalOrganisationManagementApiSpec.scala]: I don't think these tests have much value. What is useful in connector tests is to ensure that the correct query parameters are included in the http call. That is missing here.

File: app/uk/gov/hmrc/voapropertylinking/models/modernised/agentrepresentation/AppointAgent.scala (hunk 4 above)
Comment (habeeb01) [reply app/uk/gov/hmrc/voapropertylinking/models/modernised/agentrepresentation/AppointAgent.scala]: I have not implemented that. I was told that we would be joining the old journey for that scenario. I have not catered for "choose from list"

File: app/uk/gov/hmrc/voapropertylinking/models/modernised/agentrepresentation/AgentDetails.scala [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package uk.gov.hmrc.voapropertylinking.models.modernised.agentrepresentation
+
+import play.api.libs.json.{Json, OFormat}
+
+case class AgentDetails(name: String, address: String)
+
+object AgentDetails {
+  implicit val format: OFormat[AgentDetails] = Json.format[AgentDetails]
```
Comment (darrell-plant) [inline app/uk/gov/hmrc/voapropertylinking/models/modernised/agentrepresentation/AgentDetails.scala]: implicit val format: OFormat[AgentDetails] = Json.format

File: test/uk/gov/hmrc/voapropertylinking/connectors/modernised/ExternalOrganisationManagementApiSpec.scala (hunk 5 above)
Comment (darrell-plant) [reply test/uk/gov/hmrc/voapropertylinking/connectors/modernised/ExternalOrganisationManagementApiSpec.scala]: did you consider this?

**Comment by darrell-plant** [comment]
One comment, but LGTM 👍

File: test/uk/gov/hmrc/test/AllMocks.scala [hunk 7]
```diff
@@ -53,6 +53,7 @@ trait AllMocks extends MockitoSugar { me: BeforeAndAfterEach =>
   val mockServicesConfig: ServicesConfig = mock[ServicesConfig]
   val mockVoaHttpClient: VoaHttpClient = mock[VoaHttpClient]
   val mockWSRequest: WSRequest = mock[WSRequest]
+  val mockOrganisationManagementApi: ExternalOrganisationManagementApi = mock[ExternalOrganisationManagementApi]
```
Comment (DanielConnelly) [inline test/uk/gov/hmrc/test/AllMocks.scala]: can we add this in alphabetical order, makes it easier to maintain.

File: test/uk/gov/hmrc/test/AllMocks.scala [hunk 8]
```diff
@@ -75,6 +76,7 @@ trait AllMocks extends MockitoSugar { me: BeforeAndAfterEach =>
       mockPropertyLinkingService,
       mockServicesConfig,
       mockVoaHttpClient,
-      mockWSRequest
+      mockWSRequest,
+      mockOrganisationManagementApi
```
Comment (DanielConnelly) [inline test/uk/gov/hmrc/test/AllMocks.scala]: as above.

File: test/uk/gov/hmrc/voapropertylinking/connectors/modernised/ExternalOrganisationManagementApiSpec.scala (hunk 5 above)
Comment (DanielConnelly) [reply test/uk/gov/hmrc/voapropertylinking/connectors/modernised/ExternalOrganisationManagementApiSpec.scala]: this.

**Comment by DanielConnelly** [comment]
LGTM 👍

**Pull request merged**